### PR TITLE
ops: tpetra-block: revise rank-2 `update`

### PR DESCRIPTION
**Pull request by mzuzek**
Contents:
- [x] revised overload constraints
- [x] added extent checks
- [x] added explicit type casting for coefficients
- [x] added NaN injection unit tests
- [x] covered no-alpha overload in unit tests

refs #449

### CODE REVIEW

File: include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp [hunk 1]
```diff
@@ -53,27 +53,56 @@ namespace pressio{ namespace ops{
...
-       const T & mv1, const b_t &b)
+       const T1 & mv1, const b_t &b)
 {
-  mv.update(b, mv1, a);
+  assert(::pressio::ops::extent(mv, 0) == ::pressio::ops::extent(mv1, 0));
+  assert(::pressio::ops::extent(mv, 1) == ::pressio::ops::extent(mv1, 1));
+
+  using scalar_t = typename ::pressio::Traits<T>::scalar_type;
+  scalar_t a_{a};
+  scalar_t b_{b};
```
Comment (fnrizzi) [inline include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp:82]: `const scalar_t`

Comment (mzuzek) [reply include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp]: fixed with 479098106febcd8598494ee7fe62279cfad70aac

File: include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp [hunk 2]
```diff
@@ -53,27 +53,56 @@ namespace pressio{ namespace ops{
...
+   || std::is_integral<typename ::pressio::Traits<T>::scalar_type>::value)
+  && std::is_convertible<b_t, typename ::pressio::Traits<T>::scalar_type>::value
   >
-update(T & mv, const T & mv1, const scalar_t & b)
+update(T & mv, const T1 & mv1, const b_t & b)
 {
+  using scalar_t = typename ::pressio::Traits<T>::scalar_type;
   constexpr auto zero = ::pressio::utils::Constants<scalar_t>::zero();
-  mv.update(b, mv1, zero);
+  pressio::ops::update(mv, zero, mv1, b);
```
Comment (fnrizzi) [inline include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp:105]: same comment as in the tpetra PR

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see  comment

File: include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp (hunk 2 above)
Comment (mzuzek) [reply include/pressio/ops/tpetra_block/ops_multi_vector_update.hpp]: fixed on f8e4aa5361a8b1203b86da9b130d67ad612bb53f

**Review by fnrizzi — APPROVED** [review]

**Comment by mzuzek** [comment]
@fnrizzi thank you for the review and your comments!

**Pull request merged**